.github/workflows/skms.yaml

```diff
@@ -0,0 +1,115 @@
+
+          function skms_request() {
+            local username=$1
+            local passkey=$2
+            local object=$3
+            local method=$4
+            local method_params=$5
+            local url=$6
+
+            if [ -z "$username" ]; then
+              echo "First parameter missing, must be SKMS username"
+              return 1
+            fi
+
+            if [ -z "$passkey" ]; then
+              echo "Second parameter missing, must be SKMS passkey"
+              return 1
+            fi
+
+            if [ -z "$object" ]; then
+              echo "Third parameter missing, must be an SKMS dao object"
+              return 1
+            fi
+
+            if [ -z "$method" ]; then
+              echo "Fourth parameter missing, must be SKMS dao method"
+              return 1
+            fi
+
+            if [ -z "$method_params" ]; then
+              method_params='{}'
+            fi
+
+            if [ -z "$url" ]; then
+              url=$DEFAULT_SKMS_URL
+            fi
+
+            local params="{\"_username\":\"${username}\",\"_passkey\":\"${passkey}\",\"_object\":\"${object}\",\"_method\": \"${method}\"}"
+            params=$(echo "$params $method_params" | jq -s add)
+
+            local response=$(curl --write-out "%{http_code}" --silent --output response.txt https://${url}/web_api --data-urlencode "_parameters=${params}")
+
+            if [ $response != "200" ]; then
+              echo "CURL call returned HTTP status code: $response"
+              exit 1
+            elif grep -q "\"status\":\"error\"" response.txt; then
+              echo "CMR creation failed with response: "
+              cat response.txt
+              exit 1
+            else
+              echo "CMR creation was successful"
+            fi
+          }
+
+          function sanitizeStr() {
+            local str=$1
+
+            if [ -z "$str" ]; then
+              echo "First parameter missing, must be a valid string"
+              return 1
+            fi
+
+            str="${str//\"/""}"
+            str="${str//\'/""}"
+            str="${str//$'\r\n\t'/ }"
+
+            echo "$str"
+          }
+
+          end_date=$(TZ=":America/Los_Angeles" date '+%Y-%m-%d %H:%M')
+          release_title=$(sanitizeStr "${{ github.event.pull_request.title }}")
+          release_details=$(sanitizeStr "${{ github.event.pull_request.body }}")
+
+          skms_request '${{ secrets.SKMS_USER }}' '${{ secrets.SKMS_PASS }}' CmrDao createCmrFromPreapprovedChangeModel '{"change_executor":"${{ github.actor }}","maintenance_window_end_time":"${end_date}","maintenance_window_start_time":"${start_date}","preapproved_change_model_id":"847787","summary":"Release Title:${release_title}::${release_details}::Pull Request Number:${{ github.event.pull_request.number }}::Pull Request Created at:${{ github.event.pull_request.created_at }}::Pull Request Merged at:${{ github.event.pull_request.merged_at }}"}'
```

Comment:
hardcode for preapproved_change_model_id

Reply:
@JackySun9, every team will have to create their own preapproved change model id and then add it to their site's repo for this github action.  This one is strictly for Milo so having the hardcoded value is okay.  I will be adding this to Milo College's repo once this is merged and working with Milo's repository and then the one in Milo College will be the one other sites currently on Milo and future sites will need to pull and update accordingly.  That is where I will remove the hardcoded id and leave it as a placeholder.  There really isn't a use case for a specific site repo to have different change model id's since it will fall under the same service/application that is having a change/release happening.  Each change model in the SKMS system is supposed to associate what services/apps are affected or potentially affected by a release coming from the change/deployment.

Reply:
ok